### DISCUSSION TITLE
How to add method to interface at specific place

### DISCUSSION THREAD

**Discussion by MartinMystikJonas** [rectorphp/rector#7359] [General]
In one of our old rules we add new method to processed node interface like this:
```php
    /**
     * @param Interface_ $node
     */
    public function refactor(Node $node): ?Node
          // ...
          $methodBuilder = new Method("first");
          $methodBuilder->makePublic();
          // ...
          $methodNode = $methodBuilder->getNode();
          $methodNode->stmts = null;
          $this->nodesToAddCollector->addNodeAfterNode($methodNode, $node->getMethod("fetch"));
          // ...
          return $node;
    }
```
But addNodeAfterNode is deprecated. What is correct way to do it now?

I tried inserting it into stmts like this:
```php
Arrays::insertAfter($node->stmts, array_search($node->getMethod("fetch"), $node->stmts, true), [$methodNode]);
```
but it did not work.

**Comment by TomasVotruba** [comment]
If your work with the `Interface_` node, you can add it directly:

```php
$node->stmts[] = $newMethod;
```

**Reply by MartinMystikJonas** [reply]
Well that was what I was trying to do. I just do not want to add it at the end but after related method by inserting it in the middle of stmts array. BUt it seems error was somewhere else.

**Reply by TomasVotruba** [reply]
In case you need some specific position, you can use:

```php
foreach ($node->stmts as $key => $stmt) {
    if ($this->isCorrectPosition($stmt)) {
         array_splice($node->stmts, $key, 0, [$newMethod]);
         break;
    }
}
```

We use it in places where we need to add constructor as first method, but after properties/constant.
